CONTRIBUTING.md

```diff
@@ -335,11 +335,30 @@ Mocha tests are contained in `__tests__` directories.
 #### Jest
 Jest tests are stored in the same directory as source code files with the `.test.js` suffix.
 
-### Running Jest Unit Tests
-
-```bash
-node scripts/jest
-```
+### Running Kibana tests
+
+The following table outlines possible test file locations and how to invoke them:
+
+| Test location                                                                                                                                           | Runner command (working directory is kibana root)                                       |
+| ------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------- |
+| `src/**/*.test.js`<br>`src/**/*.test.ts`                                                                                                                | `yarn test:jest -t regexp [test path]`                                                  |
+| `src/**/__tests__/**/*.js`<br>`packages/kbn-datemath/test/**/*.js`<br>`packages/kbn-dev-utils/src/**/__tests__/**/*.js`<br>`tasks/**/__tests__/**/*.js` | `node scripts/mocha --grep=regexp [test path]`                                          |
+| `**/integration_tests/**/*.test.js`                                                                                                                     | `node scripts/jest_integration -t regexp [test path]`                                   |
+| `test/*integration/**/config.js`<br>`test/*functional/**/config.js`<br>`x-pack/test/*integration/**/config.js`<br>`x-pack/test/*functional/config.js`   | `node scripts/functional_tests --config test/[directory]/config.js --grep=regexp`       |
+| `x-pack/**/*.test.js`<br>`x-pack/**/*.test.ts`                                                                                                          | `cd x-pack && yarn test:jest -t regexp [test path]`                                     |
+| `x-pack/test/*integration/**/config.ts`                                                                                                                 | ??                                                                                      |
```

Comment:
I think the value for this row can be:

```
cd x-pack && node scripts/functional_tests_server.js
node scripts/functional_test_runner.js --config x-pack/test/api_integration/config.js --grep "Some string from the test case name"
```

Reply:
We probably want to keep the X-Pack specific docs here: https://github.com/elastic/kibana/blob/master/x-pack/README.md

Reply:
SGTM. It would be nice to cross-link between those two docs, in that case.

Reply:
The problem I ran into here was how do I run functional tests if the config file is written as a typescript file as opposed to a javascript file?